torch/testing/_internal/common_nn.py

```diff
@@ -3691,7 +3694,7 @@ def kldivloss_reference(input, target, reduction='mean'):
         return result.mean()
     elif reduction == 'sum':
         return result.sum()
-    elif reduction == 'batchmean' and results.dim() != 0:
+    elif reduction == 'batchmean' and result.dim() != 0:
```

Comment:
This is clearly unused code (because the variable doesn't exist), not sure if that's intended or not. @nikitaved you added this in commit 35cdb785228b8, can you comment on how this `_reference` stuff was used?

Reply:
@rgommers , I am not sure I understand. I did not add any `_reference`, and did not touch any `reduction` options...

Reply:
@rgommers , the reference is used in the tests, I did not introduce `kldivloss_reference`, but I did introduce `kldiv_log_target_reference` which was a copy of `kldivloss_reference`.

Reply:
yes, my method suffers from the same issue.

Reply:
okay the rest is used, just not the `batchmean` part - let's leave this as is.